lib/lumberjack/server.rb

```diff
@@ -214,13 +226,14 @@ def initialize(fd)
       super()
       @parser = Parser.new
       @fd = fd
+      @closed = Concurrent::AtomicBoolean.new
```

Comment:
This is only for closing a specific connection gracefully, I don't think we need to share this state in the current context of the code.

If you close the `tcp_server` socket the client will terminate with an IOError, which will unblock the client and close the socket. At https://github.com/elastic/ruby-lumberjack/pull/13/files#diff-e02f5a53687afb84cbe78c42e99f189fR239 you see that this error will be silently ignored.